tests/Functional/TestCase.php

```diff
@@ -82,16 +86,114 @@ protected static function runFrontScript(array $args, string $cwd = __DIR__): Pr
         return $process;
     }
 
-    protected static function createFiles(array $filenames): void
+    protected static function createFiles(string $baseDir, array $filenames): void
     {
         foreach ($filenames as $filename) {
-            $dirname = dirname($filename);
-            if (!file_exists($dirname)) {
-                self::assertTrue(mkdir($dirname, 0777, true), 'Created directory.');
+            self::createFile($baseDir, $filename);
+        }
+    }
+
+    protected static function createFile(string $baseDir, string $filename): void
+    {
+        $filename = "{$baseDir}/{$filename}";
+        $dirname = dirname($filename);
+        if (!file_exists($dirname)) {
+            self::assertTrue(mkdir($dirname, 0777, true), 'Created directory.');
+        }
+        self::assertTrue(touch($filename), 'Created file.');
+        self::assertNotFalse(realpath($filename), 'Got absolute path.');
+    }
+
+    protected static function changeFile($dir, $filename): void
+    {
+        file_put_contents(
+            DirectoryUtil::ensureTrailingSlash($dir) . $filename,
+            self::CHANGED_CONTENT
+        );
+    }
+
+    protected static function deleteFile($dir, $filename): void
+    {
+        unlink(DirectoryUtil::ensureTrailingSlash($dir) . $filename);
+    }
+
+    protected static function fixSeparators(string $path)
+    {
+        return str_replace(['/', '\\'], DIRECTORY_SEPARATOR, $path);
+    }
+
+    protected static function fixSeparatorsMultiple(&...$paths): void
+    {
+        foreach ($paths as &$path) {
+            $path = self::fixSeparators($path);
+        }
+    }
+
+    protected static function assertDirectoryListing(string $dir, array $expected, string $ignoreDir = ''): void
+    {
+        $ignoreDir = DirectoryUtil::stripAncestor($ignoreDir, $dir);
+
+        $expected = array_map(static function ($path) {
```

Comment:
```suggestion
        $expected = array_map([self::class, 'fixSeparators'], $expected);
```

Reply:
What's the rationale behind this suggestion?